src/HDF5.jl

```diff
@@ -2019,20 +2031,22 @@ function hdf5_to_julia_eltype(objtype)
                     (membertypes[1] <: HDF5Scalar)
 
         if iscomplex
-            T = Complex{membertypes[1]}
+          return Complex{membertypes[1]}
         else
-            T = NamedTuple{Symbol.(membernames), Tuple{membertypes...}}
+          return NamedTuple{Symbol.(membernames), Tuple{membertypes...}}
         end
     elseif class_id == H5T_ARRAY
-        T = hdf5array(objtype)
-    else
-        error("Class id ", class_id, " is not yet supported")
+        nd = h5t_get_array_ndims(objtype.id)
```

Comment:
Delete `hdf5array` ?

Reply:
yeah deleting it probably makes sense as long as we don't think many people are using it. I don't really like the name since it seems pretty unclear exactly what to expect `hdf5array` to do.